core/domain/skill_domain.py

```diff
@@ -247,6 +322,71 @@ def __init__(
         self.last_updated = last_updated
         self.version = version
 
+    def validate(self):
+        """Validates various properties of the Skill object.
+
+        Raises:
+            ValidationError: One or more attributes of skill are invalid.
+        """
+        if not isinstance(self.description, basestring):
+            raise utils.ValidationError(
+                'Expected description to be a string, received %s'
+                % self.description)
+
+        if not isinstance(self.misconceptions_schema_version, int):
+            raise utils.ValidationError(
+                'Expected misconceptions schema version to be an integer, ' +
```

Comment:
You don't need the ` +` at the end (to join strings in python). Ditto below.

Reply:
Done.